### PR TITLE
add a rust trycp client crate

### PR DESCRIPTION
**Pull request by neonphog**
- Pulls some shared protocol types out into a trycp_api crate to be used by both the server and client.
- Adds a trycp_client crate that can be used to talk to a trycp server.

### CODE REVIEW

File: crates/trycp_api/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "trycp_api"
+version = "0.0.100-alpha0"
```
Comment (ThetaSinner) [inline crates/trycp_api/Cargo.toml:3]: We should probably start versioning these, can it just match the tryorama minor version for now @jost-s?

File: crates/trycp_client/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+license = "CAL-1.0"
+edition = "2021"
+
+[dependencies]
+futures = "0.3.30"
+rmp-serde = "1.1"
+serde_json = "1.0.117"
+tokio = { version = "1.38.0", features = [ "full" ] }
+tokio-tungstenite = "0.21"
+trycp_api = { version = "0.0.100-alpha0", path = "../trycp_api" }
```
Comment (ThetaSinner) [inline crates/trycp_client/Cargo.toml:13]: To the workspace for easy finding when updating versions? No release-automation here :)

**Review by ThetaSinner — APPROVED** [review]
Looks great!

File: crates/trycp_api/Cargo.toml (hunk 1 above)
Comment (jost-s) [reply crates/trycp_api/Cargo.toml:3]: Cool

**Review by jost-s — APPROVED** [review]
Awesome!

I suggest smoke tests for the client at some stage.

**Comment by ThetaSinner** [comment]
Yep, absolutely agree - for now Wind Tunnel is going to be using this so we'll have reasonably quick feedback if something breaks :)

**Pull request merged**